src/core/config/config_module/merge.rs

```diff
@@ -0,0 +1,362 @@
+        &self.type_of
+    }
+}
+
+impl TypedEntry for Arg {
+    fn type_of(&self) -> &crate::core::Type {
+        &self.type_of
+    }
+}
+
+trait FederatedMergeCollection:
+    IntoIterator<Item = (String, Self::Entry)>
+    + FromIterator<(String, Self::Entry)>
+    + Extend<(String, Self::Entry)>
+{
+    type Entry: TypedEntry;
+
+    fn remove(&mut self, name: &str) -> Option<Self::Entry>;
+}
+
+impl<Entry: TypedEntry> FederatedMergeCollection for IndexMap<String, Entry> {
+    type Entry = Entry;
+
+    fn remove(&mut self, name: &str) -> Option<Self::Entry> {
+        self.swap_remove(name)
+    }
+}
+
+impl<Entry: TypedEntry> FederatedMergeCollection for BTreeMap<String, Entry> {
+    type Entry = Entry;
+
+    fn remove(&mut self, name: &str) -> Option<Self::Entry> {
+        self.remove(name)
+    }
+}
+
+impl<C> FederatedMergeIntersection for C
+where
+    C: FederatedMergeCollection,
+    C::Entry: FederatedMergeIntersection,
+{
+    fn federated_merge_intersection(mut self, other: Self) -> Valid<Self, String> {
+        Valid::from_iter(other, |(name, other_field)| {
+        match self.remove(&name) {
+            Some(field) => FederatedMergeIntersection::federated_merge_intersection(field, other_field).map(|merged| Some((name.clone(), merged))),
+            None => {
+                if other_field.type_of().is_nullable() {
+                    Valid::succeed(None)
+                } else {
+                    Valid::fail("Input arg is marked as non_null on the right side, but is not present on the left side".to_string())
+                }
+            },
+        }
+        .trace(&name)
+        })
+        .fuse(Valid::from_iter(self, |(name, field)| {
+            if field.type_of().is_nullable() {
+                Valid::succeed(())
+            } else {
+                Valid::fail("Input arg is marked as non_null on the left side, but is not present on the right side".to_string()).trace(&name)
+            }
+        }))
+        .map(|(merged_fields, _)| {
+            merged_fields.into_iter().flatten().collect()
+        })
+    }
+}
+
+impl<C> FederatedMergeUnion for C
+where
+    C: FederatedMergeCollection,
+    C::Entry: FederatedMergeUnion,
+{
+    fn federated_merge_union(mut self, other: Self) -> Valid<Self, String> {
+        Valid::from_iter(other, |(name, other_field)| match self.remove(&name) {
+            Some(field) => field
+                .federated_merge_union(other_field)
+                .map(|merged| (name.clone(), merged))
+                .trace(&name),
+            None => Valid::succeed((name, other_field)),
+        })
+        .map(|merged_fields| {
+            let mut merged_fields: C = merged_fields.into_iter().collect();
+            merged_fields.extend(self);
+
+            merged_fields
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs;
+
+    use insta::assert_snapshot;
+    use tailcall_fixtures::configs::federation;
```

Comment:
we will slowly start deprecating `tailcall_fixtures` now that we have `include_config!` macro. Move the .graphQL file closer to the actual test.

Reply:
I though `tailcall_fixtures` are mostly intended to reuse fixtures around tests and include_config doesn't solve this